FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/gmailapi/passwordprotected/BaseComposeScreenPasswordProtectedDisallowedTermsTest.kt

```diff
@@ -0,0 +1,104 @@
+import com.flowcrypt.email.R
+import com.flowcrypt.email.TestConstants
+import com.flowcrypt.email.database.entity.AccountEntity
+import com.flowcrypt.email.matchers.CustomMatchers.Companion.withTextViewDrawable
+import com.flowcrypt.email.matchers.TextViewDrawableMatcher
+import com.flowcrypt.email.rules.FlowCryptMockWebServerRule
+import com.flowcrypt.email.ui.base.BaseComposeGmailFlow
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.Before
+
+/**
+ * @author Denys Bondarenko
+ */
+open class BaseComposeScreenPasswordProtectedDisallowedTermsTest(
+  accountEntity: AccountEntity = BASE_ACCOUNT_ENTITY
+) : BaseComposeGmailFlow(accountEntity) {
+  override val mockWebServerRule =
+    FlowCryptMockWebServerRule(TestConstants.MOCK_WEB_SERVER_PORT, object : Dispatcher() {
+      override fun dispatch(request: RecordedRequest): MockResponse {
+        return handleCommonAPICalls(request)
+      }
+    })
+
+  @Before
+  fun setUp() {
+    onView(withId(R.id.editTextEmailAddress))
+      .perform(
+        typeText(TestConstants.RECIPIENT_WITHOUT_PUBLIC_KEY_ON_ATTESTER),
+        pressImeActionButton(),
+        closeSoftKeyboard()
+      )
+    //need to leave focus from 'To' field. move the focus to the next view
+    onView(withId(R.id.editTextEmailSubject))
+      .perform(scrollTo(), click(), replaceText(MATCHING_SUBJECTS.first()))
+
+    onView(withId(R.id.btnSetWebPortalPassword))
+      .check(matches(isDisplayed()))
+      .check(matches(withText(getResString(R.string.tap_to_protect_with_web_portal_password))))
+      .check(
+        matches(
+          withTextViewDrawable(
+            resourceId = R.drawable.ic_password_not_protected_white_24,
+            drawablePosition = TextViewDrawableMatcher.DrawablePosition.LEFT
+          )
+        )
+      ).perform(click())
+
+    onView(withId(R.id.eTPassphrase))
+      .perform(
+        replaceText(PASSWORD),
+        closeSoftKeyboard()
+      )
+
+    onView(withId(R.id.btSetPassword))
+      .perform(click())
+  }
+
+  companion object {
+    private const val PASSWORD = "Qwerty1234@"
+    const val URL = "https://flowcrypt.com"
+
+    val MATCHING_SUBJECTS = listOf(
+      "[Classification: Data Control: Internal Data Control] Quarter results",
+      "Conference information [Classification: Data Control: Internal Data Control]",
+      "[Classification: Data Control: Internal Data Control]",
+      "Conference information [Classification: Data Control: Internal Data Control] Android"
+    )
+
+    val NON_MATCHING_SUBJECTS = listOf(
+      "[Classification: Data Control: Internal Data Control]! Quarter results",
+      "Conference information [Classification: Data Control: Internal Data Control])",
+      "([Classification: Data Control: Internal Data Control]",
+      "Conference information ([Classification: Data Control: Internal Data Control]) Android"
+    )
```

Comment:
All subjects from this list should trigger error message as they contain full string from disabled terms array. So even if found term is surrounded by brackets or other characters - it still should disallow password-protected messages.

Reply:
Got it. Just in case(because early we had `it should work only for exact matches - if we have droid in the list of strings, password-protected messages shouldn't be disabled for subjects with Android word`). Should I disallow password-protected messages for the following case:

`SomeCharactersBeforeWithOrWithoutWhitespaces[Classification: Data Control: Internal Data Control]SomeCharactersAfterWithOrWithoutWhitespaces`

?

Reply:
probably the correct way will be:
- if search term starts and ends by non-alphanumeric symbols (like `[...]`, `(...)`) then any match will disallow password-protected messages. 
❌ `SomeCharactersBeforeWithOrWithoutWhitespaces[Classification: Data Control: Internal Data Control]SomeCharactersAfterWithOrWithoutWhitespaces` subject will disallow password-protected messages
- in other cases, when search terms starts and ends with alphanumeric symbol, then only match which surrounded by non-alphanumeric will disallow password-protected messages. Like this, when `droid` is disallowed term:
❌ `[droid]` disallows
❌ `check -droid- case` disallows
✅ `Microdroid androids` allows

What do you think, won't it be too complicated to implement?
Or maybe this logic can be simplified?

cc @ioanmo226

Reply:
Sounds good!

Reply:
> What do you think, won't it be too complicated to implement?

@sosnovsky It looks Ok. I've added new changes. Please try.